app/models/product_import/reset_absent.rb

```diff
@@ -0,0 +1,33 @@
+module ProductImport
+  class ResetAbsent
+    attr_reader :products_reset_count
+
+    def initialize(entry_processor, settings, strategy)
+      @entry_processor = entry_processor
+      @settings = settings
+      @strategy = strategy
+
+      @products_reset_count = 0
+    end
+
+    # For selected enterprises; set stock to zero for all products/inventory
+    # that were not listed in the newly uploaded spreadsheet
+    def call
+      settings.enterprises_to_reset.each do |enterprise_id|
+        next unless entry_processor.permission_by_id?(enterprise_id)
+
+        strategy << enterprise_id.to_i
```

Comment:
imo creating a new operator is too smart for this simple task. A new operator means anyone that reads this code needs to understand what this operator is, etc, etc.
Why not just create a support data structure called something like authorized_enterprises and extract the logic?
I think this is easier on the brain:
```
def call
   reset_stock(authorized_enterprises(settings.enterprises_to_reset))
end

def authorized_enteprsies(all_enterprises_to_reset)
     authorized_enterprises = []
     all_enterprises_to_reset.each do |enterprise_id|
         next unless entry_processor.permission_by_id?(enterprise_id)
         authorized_enterprises << enterprise_id.to_i
     end
     authorized_enterprises
end

in reset_stock we could have a new arg authorized_enterprises and we can easily handle the empty supplier_ids.

Reply:
this also means you can get rid of the supplier_ids attribute in the strategies